pygeoapi/provider/oracle.py

```diff
@@ -54,6 +54,102 @@ class DatabaseConnection:
     """Database connection class to be used as 'with' statement.
     The class returns a connection object.
     """
+    pool = None  # Class-level connection pool
+    @classmethod
```

Comment:
there are some flake8/test errors, it's probably helpful to look into those right away because sometimes they uncover bugs